src/containers/Portfolio/Portfolio.jsx

```diff
@@ -1,109 +1,117 @@
 
-const listingProps = {
-  title: 'Portfolio',
-  listings: [
-    {
-      title: 'RAD',
-      desc: 'Rite Aid Corporation',
-      price: 200.53,
-      priceChange: 27.21,
-      amount: 20,
-    },
-    {
-      title: 'CAMT',
-      desc: 'Camtek LTD',
-      price: 4021.21,
-      priceChange: -120.23,
-      amount: 87,
-    },
-    {
-      title: 'WMT',
-      desc: 'Walmart Corporation',
-      price: 185.62,
-      priceChange: 2.45,
-      amount: 0,
-    },
-  ],
-};
+class Portfolio extends Component {
+  constructor(props) {
+    super(props);
 
-const watchListProps = {
-  title: 'Watchlist',
-  listings: [
-    {
-      title: 'ROKU',
-      desc: 'Roku Inc',
-      price: 38.93,
-      priceChange: -0.29,
-    },
-    {
-      title: 'BAC',
-      desc: 'Bank of America Corp',
-      price: 28.34,
-      priceChange: 0.68,
-    },
-  ],
-};
+    this.state = {
+      portfolioItems: [],
+      portfolioValue: '',
+      navbarLinks: [],
+      currentPortfolioName: ''
+    }
+  }
 
-const headlines = [
-  {
-    title: 'China may reject new trade talks if more tariffs imposed',
-    link: '/article1',
-  },
-  {
-    title: "Shiller: The market is experiencing 'irrational exuberance aaaaaaaaaaaaaaaaaaaaaa",
-    link: '/article2',
-  },
-  {
-    title: '3 Incredibly Cheap Technology Stocks',
-    link: '/article3',
-  },
-  {
-    title: '3 Things to Watch in the Stock Market This Week',
-    link: '/article4',
-  },
-];
+  componentDidMount() {
+    this.getAllPortfolios().then(() => this.getCurrentPortfolio());
+  }
 
-const Portfolio = () => (
-  <div className="Portfolio">
-    <Navbar
-      links={[
-        {
-          title: 'Month League',
-          location: '/league/monthLeague',
-        },
-        {
-          title: 'Penny Stocks',
-          location: '/league/pennyStocks',
-        },
-        {
-          title: 'Swing Stocks',
-          location: '/league/swingStocks',
-        },
-      ]}
-    />
-    <div className="portfolio-circle" />
-    <Graph
-      isLoading={false}
-      labels={['1/13/2018', '1/14/2018', '1/15/2018', '1/16/2018', '1/17/2018', '1/18/2018']}
-      data={[14.21, 23.21, 20.53, 19.23, 15.67, 16.23]}
-    />
-    <div className="portfolio-listing-news-area">
-      <div className="portfolio-listing-area">
-        <Listing {...listingProps} />
-        <Listing {...watchListProps} />
-      </div>
-      <div className="portfolio-news-area">
-        <News headlines={headlines} />
+  componentDidUpdate(prevProps) {
+    if (prevProps.portfolios !== this.props.portfolios) {
+      const navbarLinks = this.props.portfolios.map(portfolio => {return {title: portfolio.name}});
+      this.setState({navbarLinks});
+    }
+
+    if (prevProps.currentPortfolioId !== this.props.currentPortfolioId) {
+      const currentPortfolio = this.getPortfolioById(this.props.portfolios, this.props.currentPortfolioId);
+      this.setState({currentPortfolioName: currentPortfolio.name});
```

Comment:
Would this be setting state twice? (if portfolios and current portfolio id both change)? Not sure if really necessary but you can make it only set state once by making the if statement accept both cases and set the state once